### PR TITLE
pull, at import stage: replace warning with info report

### LINKED ISSUE #2685: pull: wrong warning on pulling import stage
Script to reproduce:

```
mkdir test
dvc import https://github.com/iterative/example-get-started model.pkl
git add .
git commit -am "init"
cd ..
git clone test test1
cd test1
dvc pull
```

outputs only:

```
WARNING: DVC-file 'model.pkl.dvc' is locked. Its dependencies are not going to be checked out.
```

I think there should be no warning, especially considering that it has done the job - pulled the `model.pkl`. Like we discussed in #2667 there should be some statistics instead.

### PR DESCRIPTION
**Pull request by sadielbartholomew**
Replace generic locked checkout warning with an information report on return when 'dvc pull' is run at import stage.

Related to #2685.

* [x] Have you followed the guidelines in our [Contributing document](https://dvc.org/doc/user-guide/contributing/core)?

* [x] ~~Does your PR affect documented changes or does it add new functionality
      that should be documented?~~ [It does not for both cases.]

-----

The new output, after the minimal steps to recreate as outlined in the Issue to close, is:

```console
$ dvc pull
Data retrieved successfully from DVC remote storage.
```

### CODE REVIEW

**Comment by shcheklein** [comment]
@sadielbartholomew thanks! let's for now just start with something simple like `Data retrieved successfully from DVC remote storage.` That additional report goes too away from what we have for other commands, also, I'm not sure that it's that useful to repeat this information (that you should know anyway). What would be useful is to show some basic stats - how many files, directories, size, etc. And it can be part of the message, not a separate table.

For the warning itlsef. It looks like it's being used in `dvc checkout` (and `dvc pull` is using the same set of functions internally), is it correct? Then either I'm missing something or I don't understand why do we show this warning at all while actually performing checkout?

**Comment by sadielbartholomew** [comment]
Thanks for the clarification @shcheklein, it makes it easier to know more specifically what you want to resolve the Issue.

> That additional report goes too away from what we have for other commands, also, I'm not sure that it's that useful to repeat this information (that you should know anyway). What would be useful is to show some basic stats - how many files, directories, size, etc. And it can be part of the message, not a separate table.

Sure, that makes sense. I read through #2667 as referenced in the Issue but was not sure what sort of "stats" would be useful. I'll squash in a new commit to change the report according to your requests above, & using those three stats you suggest. I can always add more afterwards if you think it would help.

> For the warning itlsef. It looks like it's being used in dvc checkout (and dvc pull is using the same set of functions internally), is it correct?

Yes, the warning does emerge from the call to ``checkout`` (I am sure because there are three locations containing that warning from a ``grep`` & I added some random text to each in testing to trace it).

> Then either I'm missing something or I don't understand why do we show this warning at all while actually performing checkout?

So, you prefer to just remove the warning from ``checkout`` altogether? From the context I read I thought it might need to be left in for other usage of ``checkout``, but I'm more than happy to change to remove it completely.

File: dvc/repo/checkout.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def _checkout(
         total=total, unit="file", desc="Checkout", disable=total == 0
     ) as pbar:
         for stage in stages:
-            if stage.locked:
+            if stage.locked and warn:
```
Comment (efiop) [inline dvc/repo/checkout.py]: We can actually delete this warning, as it is a leftover from some old logic that is no longer in place.

Comment (sadielbartholomew) [reply dvc/repo/checkout.py]: I see, thanks. I'll get it deleted in my next commit.

File: dvc/repo/pull.py [hunk 2]
```diff
@@ -20,11 +25,16 @@ def pull(
...
     )
     self._checkout(
-        targets=targets, with_deps=with_deps, force=force, recursive=recursive
+        targets=targets,
+        with_deps=with_deps,
+        force=force,
+        recursive=recursive,
+
     )
+    logger.info("Data retrieved successfully from DVC remote storage.")
```
Comment (shcheklein) [inline dvc/repo/pull.py]: qq - can we analyze results of self._fetch (processed_files_count in this case) and self._checkout to show a different message if nothing has changed or there are no files to process?

**Comment by sadielbartholomew** [comment]
I've just reverted the PR proposed code changes to the state where the warning has been removed for ``dvc checkout`` & where ``dvc pull`` returns (& logs as info) the message "Data retrieved successfully from DVC remote storage.", but I haven't yet had time to add in the stats. I'll get those added tomorrow, & if not at the weekend. Thanks for your patience.

File: dvc/repo/pull.py (hunk 2 above)
Comment (sadielbartholomew) [reply dvc/repo/pull.py]: Yes, sure. Sorry I was delayed in writing up the message below to say I will start adding in the stats information tomorrow, as don't have time tonight but wanted to address the first set of comments to being with. Thanks again for your feedback.

**Comment by shcheklein** [comment]
@sadielbartholomew absolutely! and thank for doing this 🙏

**Comment by efiop** [comment]
@sadielbartholomew Sorry for jumping in, but I've decided to adjust your PR a bit to get that warning removal quickly merged :) Please open a new PR for #2685 Also, I've noticed that you had your patch formatting broken, please refer to our contrib guide and install pre-commit hooks so they handle that automatically. Thanks! :slightly_smiling_face:

**Comment by sadielbartholomew** [comment]
> @sadielbartholomew Sorry for jumping in, but I've decided to adjust your PR a bit to get that warning removal quickly merged :)

No problem, thanks. Also, sorry, if I had known there was need to get the warning removed quickly I would have prioritised this over my other (non-work) dev.

> Please open a new PR for #2685 Also, I've noticed that you had your patch formatting broken, please refer to our contrib guide and install pre-commit hooks so they handle that automatically. Thanks! 🙂

Sure, I'll heed that advice & try to get the PR in shortly (though I might leave it until next week now there is less rush, without an open PR taking up reviewer time & effort).

**Comment by efiop** [comment]
@sadielbartholomew No rush at all! Thank you so much for contributing! :pray:

**Comment by sadielbartholomew** [comment]
No problems, thanks for being so helpful & welcoming :)

**Pull request merged**